src/app/SlaveInterface/SlaveInterface.c

```diff
@@ -0,0 +1,88 @@
+#include "HAL_SlaveChips.h"
+
+#include "FaultManager.h"
+#include "SlaveInterface.h"
+
+/**
+ * Get the latest voltage and draining information from the slave chips.
+ * Set faults if communication errors occur.
+ * battery_model [out] - destination for updated cell info
+ */
+void SlaveInterface_read_cell_info(BatteryModel_t* battery_model)
+{
+    float voltages[NUM_SERIES_CELLS];
+    bool is_draining[NUM_SERIES_CELLS];
+    
+    // get data from slave boards
+    Error_t err = HAL_SlaveChips_get_all_cell_data(voltages, is_draining, NUM_SERIES_CELLS);
+
+    // check for comm errors
+    if (err.active)
+    {
+        FaultManager_set_fault_active(FaultCode_SLAVE_COMM_CELLS, err.data, false);
```

Comment:
Should we be latching this error? I feel like it might be useful to have a watchdog or something. Bc not being able to communicate to a single cell once isn't a problem, but if it's continuous then it would be a problem

Reply:
If we latch the fault then it will never go away. DriveMonitor will decide how long this fault must be active to trigger a shutdown.